models/discrete/populations.py

```diff
@@ -178,70 +223,106 @@ def __init__(self, operators: list = None):
+        return f"{self.description}: d[{self.variable.symbol}]/dt = {self.equation} in variables '{self.variables.get_symbols()}' and parameters {self.parameters.get_symbols()}"
+
+    @staticmethod
+    def _get_dependencies(equation, variables: Variables, parameters: Parameters, warn = True):
+        '''
+        Returns the sublists of variables and parameters whose symbols appear as free symbols of self.equation
+
+        Args:
+            variables (Variables)
+            parameters (Parameters)
+            warn (bool): raise an error if there are symbols not accounted for
+        '''
+        variable_symbols = set(variables.get_symbols())
+        parameter_symbols = set(parameters.get_symbols())
+        equation_symbols = sym.sympify(equation, locals=sym_custom_ns).free_symbols
+        if warn and not equation_symbols.issubset(variable_symbols.union(parameter_symbols)):
+            print("Unaccounted symbols in equation")
+        equation_variables = [variables._objectify(symbol) for symbol in equation_symbols.intersection(variable_symbols)]
+        equation_parameters = [parameters._objectify(symbol) for symbol in equation_symbols.intersection(parameter_symbols)]
+        return Variables(equation_variables), Parameters(equation_parameters)
 
     def _check_equation_completeness(self):
-        variable_symbols = self.all_variables.get_symbols()
+        variable_symbols = self.variables.get_symbols()
         parameter_symbols = self.parameters.get_symbols()
         equation_symbols = self.equation.free_symbols
         difference = equation_symbols - set(variable_symbols + parameter_symbols)
         if difference:
-            raise Exception(f"The {'symbol' if len(difference) == 1 else 'symbols'} '{difference}' in equation '{self.equation}' {'does' if len(difference) == 1 else 'do'} not have an associated {Variable.__name__} object.")
-    
+            raise Exception(
+                f"The {'symbol' if len(difference) == 1 else 'symbols'} '{difference}' in equation '{self.equation}' {'does' if len(difference) == 1 else 'do'} not have an associated {Variable.__name__} object."
+            )
+
     def get_variables(self):
-        return [v.symbol for v in self.all_variables]
-    
+        return [v.symbol for v in self.variables]
+
     def get_parameters(self):
         return [p.symbol for p in self.parameters]
-    
+
 
 class Operator:
-    '''
+    """
     An Operator object will define generic update rules for a Variables object, in the same way that an Update_Rule defines
     an update rule for a Variable object.
-    '''
+    """
+
     def __init__(self, name, function, description):
         self.name = name
         self.operator = function
         self.description = description
 
-'''
+
+"""
 Population classes
-'''
+"""
+
+T = sym.Symbol("T")
+time = SimVariable(Variable(T, 0, "system time"))
```

Comment:
I don't think this definition of `time` is needed globally.
I only see it used within the definition of System, so it can be defined locally there.

Reply:
At least the symbol 'T' needs to be available for every population. My hope here is that variable and symbol will become synonymous, that's all.